source/common/stats/symbol_table_impl.h

```diff
@@ -466,7 +466,8 @@ class StatName {
    * hasher and comparator.
    */
   absl::string_view dataAsStringView() const {
-    return {reinterpret_cast<const char*>(data()), dataSize()};
+    return {reinterpret_cast<const char*>(data()),
+            static_cast<absl::string_view::size_type>(dataSize())};
```

Comment:
Isn't the underlying problem the abuse of uint64_t in line 409 above for dataSize()?

There's a significant penalty for using a 64 bit int in 32 bit contexts.

Reply:
@jmarantz ?

Reply:
I don't mind changing the underlying type of `dataSize`, but I don't really have any context on the original choice.

Reply:
Personally I'd prefer size_t for this; IIUC Envoy style has veered toward using uint64_t. @mattklein123 may have more context.

I'm fine with either:
 * this change, which makes sense in the context of returning a string_view
 * changing dataSize() et al to return size_t, which makes sense generally IMO

I slightly prefer the latter but I'm OK with this. @mattklein123 what do you prefer?

Reply:
Sure we can use size_t. Typically we have used explicit types for things that can wrap, but for something like this size_t seems fine.

/wait

Reply:
`uint64_t` seems to be used extensively here to represent a number of bytes. `dataSize()` is also called in a number of locations. Just to clarify, is the proposal to switch `size_t` for all of these cases?

Reply:
Yes size_t seems appropriate for number of bytes.

Reply:
Despite updating the type used for representing byte counts throughout the file, I left this cast in; I don't know what sort of contract `absl::string_view::size_type` has with respect to `size_t`.

If folks feel it's redundant though now, I can remove it.